pkgs/development/compilers/sagittarius-scheme/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, fetchurl
+, cmake
+, pkgconfig
+, libffi
+, boehmgc
+, openssl
+, zlib
+, odbc ? true
+, libiodbc
+}:
+
+let version = "0.9.6";
+    odbcDeps = if odbc then [ libiodbc ] else [];
+in stdenv.mkDerivation {
+  name = "sagittarius-scheme-${version}";
+  src = fetchurl {
+    url = "https://bitbucket.org/ktakashi/sagittarius-scheme/downloads/sagittarius-${version}.tar.gz";
+    sha256 = "03nvvvfd4gdlvq244zpnikxxajp6w8jj3ymw4bcq83x7zilb2imr";
+  };
+  preBuild = ''
+           export LD_LIBRARY_PATH="$(pwd)/build"
+  '';
+  nativeBuildInputs = [ pkgconfig cmake ];
+
+  buildInputs = builtins.concatLists [[ libffi boehmgc openssl zlib ]  odbcDeps];
+
+  meta = with stdenv.lib; {
+    description = "Sagittarius Scheme is an R6RS/R7RS Scheme system";
+    longDescription = ''
+    Sagittarius Scheme is a free Scheme implementation supporting
+    R6RS/R7RS specification.
+
+    Features:
+
+     -  Builtin CLOS.
+     -  Common Lisp like reader macro.
+     -  Cryptographic libraries.
+     -  Customisable cipher and hash algorithm.
+     -  Custom codec mechanism.
+     -  CL like keyword lambda syntax (taken from Gauche).
+     -  Constant definition form. (define-constant form).
+     -  Builtin regular expression
+     -  mostly works O(n)
+     -   Replaceable reader
+    '';
+    homepage = https://bitbucket.org/ktakashi/sagittarius-scheme;
+    license = licenses.bsd2;
+    platforms = platforms.x86;
```

Comment:
After briefly reading the `homepage` it seems `platforms.all` would be more appropriate.

Reply:
Done.